### PR TITLE
Domains: A/B test only one primary button on domain suggestions

### PR DESCRIPTION
**Pull request by sixhours**
#### Changes proposed in this Pull Request

* This test gives only the best matched domain a primary button; the best alternative gets a secondary button, and the rest of the suggestions get borderless buttons. This helps give the information a visual hierarchy.

**allPrimary (control):**

screen shot 2019-01-10 at 2 39 10 pm

**onePrimary:**

image

#### Testing instructions

* Switch to this PR and navigate to `/domains/add` or `/start/domains`
* Assign yourself to the `allPrimary` variation of the `domainSearchButtonStyles` test
* Check that all the buttons on the domain suggestions are primary
* Switch to the `onePrimary` variation
* Check that only the best matched domain is primary and the rest are secondary buttons

### CODE REVIEW

**Comment by sixhours** [comment]
The tests are failing on accessibility due to the way the search results page is laid out; wrapper divs with `onClick` events are not tab-able, but the buttons within each div are. If we make the divs tab-able, it makes the page harder to navigate (twice as many things to tab).  I think we can ignore these warnings for now.

**Comment by fditrapani** [comment]
Nice update! A lot easier on the eyes for sure. 

cc @drw158 and @cburton4

**Comment by blowery** [comment]
>  I think we can ignore these warnings for now.

Could we please add some [eslint-ignore directives](https://eslint.org/docs/user-guide/configuring#disabling-rules-with-inline-comments) if we're ok with the violations? It'll help make it clear for future folks working on the file that these things were already considered.

**Comment by hambai** [comment]
Also please check out p99Zz8-bU-p2 and p989we-cM-p2 as we've moved in the exact opposite direction before

**Comment by scruffian** [comment]
I created a PR to test this last week: https://github.com/Automattic/wp-calypso/pull/30018

**Comment by aidvu** [comment]
@scruffian your test is for the featured suggestions (primary) flag. There is another change here which moves from buttons to borderless buttons in the suggestions list below, so we need one for that too.

**Comment by drw158** [comment]
Sorry for the missed ping. I think this is a wonderful update and follows the primary button guidelines.

Are the domain list rows clickable, or is the Select text button the only thing that's clickable? If the whole row is clickable, then I think the change is great, if not, then I might consider making the Select buttons default buttons instead. It's more repetitive and noisy, but the text buttons may be too easy to miss.

**Comment by danhauk** [comment]
I think this could be a good improvement on the UX of domain search, especially as we move towards unifying experiences with the Muriel system. It can help to pull focus to one domain and make the decision point easier for folks.

However, I do think it should be implemented as a test. It has the potential to have a big impact on conversion rates.

When implementing the redesign we didn't test the change from borderless buttons to primary buttons, so there's no quantitative data available for how the changes in buttons might affect conversion.

**Comment by danhauk** [comment]
I've updated this PR to include the A/B test. Please go through and test the two variations per the updated instructions in the initial comment.

See p99Zz8-w6-p2 for more test details.

**Comment by fditrapani** [comment]
Looks good with the buttons!

File: client/components/domains/featured-domain-suggestions/index.jsx [hunk 1]
```diff
@@ -112,6 +112,7 @@ export class FeaturedDomainSuggestions extends Component {
 						isSignupStep={ this.props.isSignupStep }
 						uiPosition={ 0 }
 						fetchAlgo={ this.getFetchAlgorithm( primarySuggestion ) }
+						buttonStyles={ { primary: true } }
```
Comment (aidvu) [inline client/components/domains/featured-domain-suggestions/index.jsx]: Don't think we need this. If we can derive these styles just from the `isFeatured` flag, I'd rather we do that in the component.

File: client/components/domains/domain-registration-suggestion/index.jsx [hunk 2]
```diff
@@ -217,6 +219,11 @@ class DomainRegistrationSuggestion extends React.Component {
 			suggestion: { domain_name: domain, product_slug: productSlug, cost },
 		} = this.props;
 
+		let buttonStyles = { primary: true };
+		if ( abtest( 'domainSearchButtonStyles' ) === 'onePrimary' ) {
+			buttonStyles = ! isFeatured ? {} : this.props.buttonStyles;
```
Comment (aidvu) [inline client/components/domains/domain-registration-suggestion/index.jsx]: If we go by the comment below, it's probably a good idea to just reset styles here if not featured.

Comment (aidvu) [reply client/components/domains/domain-registration-suggestion/index.jsx]: Disregard, missed the `Secondary` featured.

**Pull request merged**